From 4634c541db08e46e785c3ab788800f80a5cd65f9 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Sat, 26 Aug 2017 15:51:16 +0200
Subject: [PATCH 3/4] packages: rework workaround patch

- also while at it added command to wipe the package cache too
- also added --noprogressbar and --nointegrity to pacman ARGS
---
 src/modules/packages/main.py | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/src/modules/packages/main.py b/src/modules/packages/main.py
index 93fcc9eb7..227712fa5 100644
--- a/src/modules/packages/main.py
+++ b/src/modules/packages/main.py
@@ -67,8 +67,9 @@ class PackageManager:
             else:
                 pacman_flags = "-Sy"
 
-            check_target_env_call(["pacman", pacman_flags,
-                                   "--noconfirm"] + pkgs)
+            check_target_env_call(["pacman", pacman_flags, "--noconfirm",
+                                   "--noprogressbar", "--nointegrity"] + pkgs)
+            check_target_env_call(["pacman", "-Scc", "--noconfirm"])
         elif self.backend == "portage":
             check_target_env_call(["emerge", "-v"] + pkgs)
         elif self.backend == "entropy":
@@ -154,13 +155,8 @@ def run_operations(pkgman, entry):
     for key in entry.keys():
         entry[key] = subst_locale(entry[key])
         if key == "install":
-            if isinstance(entry[key], list):
-                for package in entry[key]:
-                    pkgman.run(package["pre-script"])
-                    pkgman.install([package["package"]])
-                    pkgman.run(package["post-script"])
-            else:
-                pkgman.install(entry[key])
+            # not sure what changed but only that work for now
+            pkgman.install(entry[key])
         elif key == "try_install":
             # we make a separate package manager call for each package so a
             # single failing package won't stop all of them
-- 
2.14.1

